src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx

```diff
@@ -59,6 +60,21 @@ export const CharactersPage: VFC = () => {
     return groupedCharacters;
   };
 
+  const [canFit4OrMore, canFitOnly3, canFitOnly2] = useMediaQuery(
+    ['(min-width: 1564px)', '(min-width: 1280px)', '(min-width: 1024px)'],
+  );
+
+  let columnCount = 4;
+  if (canFit4OrMore) {
+    columnCount = 4;
+  } else if (canFitOnly3) {
+    columnCount = 3;
+  } else if (canFitOnly2) {
+    columnCount = 2;
+  } else {
+    columnCount = 1;
+  }
```

Comment:
You can avoid `let` and `else if` here if you move out this code part as function